File: generate-logo.js
#!/usr/bin/env node

import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));

// Try to use canvas if available
try {
  const { createCanvas } = await import('canvas');

  // Create 512x512 canvas for high quality
  const canvas = createCanvas(512, 512);
  const ctx = canvas.getContext('2d');

  // Transparent background
  ctx.clearRect(0, 0, 512, 512);

  // Draw shield outline (blue)
  ctx.strokeStyle = '#2badee';
  ctx.lineWidth = 18;
  ctx.lineJoin = 'round';
  ctx.lineCap = 'round';

  // Shield path
  ctx.beginPath();
  ctx.moveTo(256, 40);
  ctx.lineTo(112, 104);
  ctx.lineTo(112, 216);
  ctx.quadraticCurveTo(256, 408, 256, 408);
  ctx.quadraticCurveTo(256, 408, 400, 216);
  ctx.lineTo(400, 104);
  ctx.closePath();
  ctx.stroke();

  // Draw X (red)
  ctx.strokeStyle = '#ff4757';
  ctx.lineWidth = 20;

  // First diagonal
  ctx.beginPath();
  ctx.moveTo(192, 192);
  ctx.lineTo(320, 320);
  ctx.stroke();

  // Second diagonal
  ctx.beginPath();
  ctx.moveTo(320, 192);
  ctx.lineTo(192, 320);
  ctx.stroke();

  // Save as PNG
  const buffer = canvas.toBuffer('image/png');
  fs.writeFileSync(path.join(__dirname, 'public', 'logo.png'), buffer);

  // Also create 32x32 favicon
  const faviconCanvas = createCanvas(32, 32);
  const faviconCtx = faviconCanvas.getContext('2d');
  faviconCtx.clearRect(0, 0, 32, 32);

  // Scale down the logo
  faviconCtx.drawImage(canvas, 0, 0, 512, 512, 0, 0, 32, 32);

  const faviconBuffer = faviconCanvas.toBuffer('image/png');
  fs.writeFileSync(path.join(__dirname, 'public', 'favicon.png'), faviconBuffer);

  console.log('✓ Logo PNG generated successfully!');
  console.log('✓ Favicon PNG generated successfully!');

} catch (err) {
  console.log('Canvas library not available. Using SVG directly.');
  console.log('\nTo generate PNG from SVG, use one of these methods:');
  console.log('\n1. Online converter: https://cloudconvert.com/svg-to-png');
  console.log('   - Upload: public/logo-hires.svg');
  console.log('   - Download as PNG');
  console.log('\n2. Command line (requires ImageMagick):');
  console.log('   convert -background none -density 300 public/logo-hires.svg -resize 512x512 public/logo.png');
  console.log('\n3. Install canvas and run this script:');
  console.log('   npm install canvas');
  console.log('   node generate-logo.js');
}

File: generate-logo.py
#!/usr/bin/env python3
"""Generate PNG logo from SVG using PIL"""

import os
import sys

try:
    from PIL import Image, ImageDraw

    # Create 512x512 image with transparent background
    img = Image.new('RGBA', (512, 512), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)

    # Shield outline (blue: #2badee)
    blue = (43, 173, 238, 255)
    red = (255, 71, 87, 255)

    # Draw shield outline
    # Shield path: top point, left side, bottom left, bottom center, bottom right, right side
    shield_points = [
        (256, 40),      # top
        (112, 104),     # top left
        (112, 216),     # left side
        (256, 408),     # bottom center
        (400, 216),     # right side
        (400, 104),     # top right
    ]

    draw.polygon(shield_points, outline=blue, width=18)

    # Draw X (red)
    # First diagonal
    draw.line([(192, 192), (320, 320)], fill=red, width=20)
    # Second diagonal
    draw.line([(320, 192), (192, 320)], fill=red, width=20)

    # Save as PNG
    img.save('public/logo.png')

    # Create 32x32 favicon
    favicon = img.resize((32, 32), Image.Resampling.LANCZOS)
    favicon.save('public/favicon.png')

    print('[OK] Logo PNG generated successfully!')
    print('[OK] Favicon PNG generated successfully!')

except ImportError:
    print('PIL not available. Please install it:')
    print('  pip install Pillow')
    print('')
    print('Or use an online converter:')
    print('  https://cloudconvert.com/svg-to-png')
    sys.exit(1)

